### DISCUSSION TITLE
Zooz ZEN30 switch contains a duplicate switch and light entity

### DISCUSSION THREAD

**Discussion by jaygorrell** [zwave-js/zwave-js#6589] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?
Adding a Zooz ZEN30 switch results in both a duplicated light and switch entity.

![image](https://github.com/zwave-js/node-zwave-js/assets/12174604/1ae90799-9e18-475e-8b60-251df4bfa083)

What did you expect to happen instead?
I'd expect to see a single switch and a single light.

  1. Add a new ZEN30 to HA

Anything else we should know?
I have 3x ZEN30 devices and a handful of ZEN71 and ZEN72 devices. The ZEN30 is the only one that has this issue. After opening a thread on the forums [here](https://community.home-assistant.io/t/zooz-devices-with-duplicate-entities/666883), the following was noted:
> The ZEN30 entities are duplicated because Zooz made a poor design decision related to multi-channel endpoints. https://github.com/zwave-js/node-zwave-js/pull/4736
>
> However there should only be a [duplicate switch](https://github.com/zwave-js/node-zwave-js/issues/4724#issuecomment-1163179279) entity, not a second light one, so maybe they fixed the problem in this latest revision of the device and the workaround is not required. You could [submit an issue](https://github.com/zwave-js/node-zwave-js/issues/new/choose) to node-zwave-js, including the full driver debug logs of a re-interview, and/or disable the entities that aren’t useful to you (firmware changelog doesn’t indicate anything).

### Software versions

Driver (node-zwave-js): 12.3.0

Home Assistant Z-Wave Integration: 1.33.0

### Device information

Manufacturer: Zooz
Model name: ZEN30
Node ID: 21

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

[home-assistant_zwave_js_2024-01-06T04-17-17.737Z.log](https://github.com/zwave-js/node-zwave-js/files/13848364/home-assistant_zwave_js_2024-01-06T04-17-17.737Z.log)

**Comment by AlCalzone** [comment]
This seems to be the new 800 series version of the ZEN30. According to the log, it looks like Endpoint 1 is the dimmer, Endpoint 2 is the on/off switch, and endpoint 0 duplicates both. So they did change how it behaves.

@jaygorrell can you confirm if both dimmers control the same device functionality and both on/off switches control the same device functionality? Which ones gets updated when you control the device manually? Can you make another log of that?

**Reply by jaygorrell** [reply]
> This seems to be the new 800 series version of the ZEN30.

Correct!

> @jaygorrell can you confirm if both dimmers control the same device functionality and both on/off switches control the same device functionality? 

This is a bit hard to explain so I'll list them explicitly. The last one here is notable.
- Dimmer & Dry Contact Relay `switch`: coupled with (2) `switch`
- Dimmer & Dry Contact Relay `light`: coupled with (1) `light`
- (1) `light`: coupled with Dimmer & Dry Contact Relay `light`
- (2) `switch`: operates independently, does not toggle any other controls

> Which ones gets updated when you control the device manually?

The scene 001 / 002 (light paddles) toggle both `light` entities. Those seem to always be coupled so I'm not sure how to check beyond that. My suspicion is that (1) `light` in my case is the more relevant one just based on the next bit:

The scene 003 (on/off switch) toggles only the (2) `switch` here. My Dimmer & Dry Contact Relay `switch` remains untouched. This aligns with using HA to control (2) as well, as it's the only entity that operates independently.

I'll have to get back to you on the log if needed.

**Reply by AlCalzone** [reply]
Yeah I'm going to need a log to figure that out. Please control each of the 4 controls with about 10 seconds of pause in between, and note the timestamps when you did what.
Then do the same for the physical controls.

**Reply by jaygorrell** [reply]
Here you go.

Here's my current UI and I went top-to-bottom before moving on to the physical switch.
![image](https://github.com/zwave-js/node-zwave-js/assets/12174604/d392ebe1-de2f-4d1b-b2c3-a9664098add1)

I started at 8:36 and waited 30 seconds between each action, give or take a couple seconds. This is Node 23.
```
8:36:00 - UI, Office Switch (`switch`) ON // turns on both switch entities
8:36:30 - UI, Office Switch (`switch`) OFF // turns off both switch entities
8:37:00 - UI, Office Switch (`light`) ON // turns on both light entities
8:37:30 - UI, Office Switch (`light`) OFF // turns off both light entities
8:38:00 - UI, (1) (`light`) ON // turns on both light entities
8:38:30 - UI, (1) (`light`) OFF // turns off both light entities
8:39:00 - UI, (2) (`switch`) ON // turns on only this one
8:39:30 - UI, (2) (`switch`) OFF // turns off only this one
8:40:00 - Physical, Top Paddle // turns on both light entities
8:40:30 - Physical, Bottom Paddle // turns off both light entities
8:41:00 - Physical, On/Off Switch // toggles only (2) switch
```

[home-assistant_zwave_js_2024-06-07T12-41-10.881Z.log](https://github.com/user-attachments/files/15739230/home-assistant_zwave_js_2024-06-07T12-41-10.881Z.log)

**Reply by AlCalzone** [reply]
Thanks. Looks like the endpoints are now working as they should and we can drop the workarounds for this revision of the device.

**Comment by AlCalzone** [comment, marked as answer]
Escalated to an issue: https://github.com/zwave-js/node-zwave-js/issues/6910
